src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,28 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getAge() == null || user.getPassword() == null) {
+            throw new NullPointerException("Incorrect password or user age");
+        }
+        if (user.getAge() < 18 || user.getPassword().length() < 6) {
+            throw new InvalidDataException("Incorrect password or user age");
+        }
+        if (user.getLogin() == null) {
+            throw new NullPointerException("Incorrect password or user age");
+        }
+        for (User person : Storage.people) {
```

Comment:
You should not get people from storage directly here, we have this only ability for test proposes.
Use storageDao.get(login) for checking if we already have a registered user with such a login